packages/product/src/repositories/product.ts

```diff
@@ -121,7 +125,34 @@ export class ProductRepository extends AbstractBaseRepository<Product> {
       return manager.create(Product, product)
     })
 
-    await manager.persist(products)
+    await manager.persistAndFlush(products)
+
+    return products
+  }
+
+  async update(
+    data: WithRequiredProperty<ProductServiceTypes.UpdateProductDTO, "id">[],
+    context: Context = {}
+  ): Promise<Product[]> {
+    const manager = (context.transactionManager ??
+      this.manager_) as SqlEntityManager
+
+    const products = await Promise.all(
+      data.map(async (updateData) => {
+        if (!isDefined(updateData.id)) {
+          throw new MedusaError(
+            MedusaError.Types.NOT_FOUND,
+            `Cannot update product without id`
+          )
+        }
+
+        const product = await manager.findOneOrFail(Product, updateData.id);
+
+        return manager.assign(product, updateData, { updateNestedEntities: true, mergeObjects: true })
+      })
+    )
+
+    await manager.persistAndFlush(products)
```

Comment:
todo: same

Reply:
Just pointing out, this is marked resolved but not fixed - in case you missed it.